### PR TITLE
Junit test results publish with multiple failure nodes

### PR DESCRIPTION
**Pull request by prawalagarwal**
Description: There are JUnit result files which may have multiple failure and error nodes for a test case. TestResult publishing task was failing with testCaseNode.failure.attributes() is not a function for these result files.

Fix: Handled the case with multiple failure nodes and publishing the first node.

Testing: Manual

### CODE REVIEW

File: src/agent/testresultreader.ts [hunk 1]
```diff
@@ -263,20 +285,41 @@ export class ResultReader {
...
                 }
             }
             else if (testCaseNode.error) {
                 outcome = "Failed";
-                if (testCaseNode.error.text) {
-                    stackTrace = testCaseNode.error.text();
+                var testNode;
+
+                if (testCaseNode.error.count() > 1) {
+                    this.command.warning("Multiple failures in the test case: " + testName + ". Picking the first for publishing!");
```
Comment (AshwiniChalla) [inline src/agent/testresultreader.ts]: do we want this to be a warning or info?

File: src/agent/testresultreader.ts [hunk 2]
```diff
@@ -263,20 +285,41 @@ export class ResultReader {
...
+                if (testNode.text) {
+                    stackTrace = testNode.text();
                 }
-                if (testCaseNode.error.attributes().message) {
-                    errorMessage = testCaseNode.error.attributes().message;
+                if (testNode.attributes().message) {
+                    errorMessage = testNode.attributes().message;
                 }
             }
             else if (testCaseNode.skipped) {
```
Comment (AshwiniChalla) [inline src/agent/testresultreader.ts]: lets have a switch case and refactoring needed.

File: src/agent/testresultreader.ts (hunk 1 above)
Comment (AshwiniChalla) [inline src/agent/testresultreader.ts]: Lets refactor these. repeating code.

**Comment by AshwiniChalla** [comment]
approved with comments

File: src/test/publishertests.ts [hunk 3]
```diff
@@ -24,10 +25,11 @@ describe('PublisherTests', function() {
         releaseUri: "abc",
         releaseEnvironmentUri: "xyz"
     };  
-
-    var readerJUnit = new trr.JUnitResultReader();    
-    var readerNUnit = new trr.NUnitResultReader();
-    var readerXUnit = new trr.XUnitResultReader();
+	
+	var command : cm.ITaskCommand;
```
Comment (allendm-msft) [inline src/test/publishertests.ts]: do we have a test covering this case already?
fix the indentation too

File: src/agent/testresultreader.ts (hunk 1 above)
Comment (allendm-msft) [inline src/agent/testresultreader.ts]: info should be good

**Comment by allendm-msft** [comment]
task version should be bumped up too?
Add the right test cases
Approved - as long as all comments are addressed

File: src/agent/testresultreader.ts (hunk 2 above)
Comment (prawalagarwal) [inline src/agent/testresultreader.ts]: We cannot have a switch here. We are not checking the state of a single entity. The first if checks if there are failure nodes, the second should be executed only if there are no failures and is thus checking error nodes. Similarly third should be executed only when there are no failure and error nodes.

File: src/agent/testresultreader.ts (hunk 1 above)
Comment (prawalagarwal) [inline src/agent/testresultreader.ts]: Changed to info. This is actually not repeating code. Explanation below

**Comment by prawalagarwal** [comment]
Task Version doesn't need to be bumped up. Test added for the new format.

**Pull request merged**